File: Q31 longest common substring.py
class Solution:
    def longestCommonPrefix(self, arr):
        if not arr:
            return ""

        arr.sort()
        a = arr[0]
        b = arr[-1]
        i = 0
        ans = []

        while i < len(a):
            if a[i] == b[i]:
                ans.append(a[i])
                i += 1
            else:
                break

        return ''.join(ans)

File: Q31 Longest common substring.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& arr) {
          sort(arr.begin(),arr.end());
        
        string a = arr[0];
        string b = arr[arr.size()-1];
        int i = 0;
        string ans="";
        char ch;
        while(i<a.size() ){
            if( a[i] == b[i]){
                ans+=a[i];
                i++;
            }
            else break;
        }
        if(ans.size() == 0)return "";
        else 
        return ans;
    }
};
